doc/developer/design/20220214_create_cluster.md

```diff
@@ -0,0 +1,191 @@
+
+show_cluster_stmt: SHOW CLUSTERS [{ LIKE 'pattern' | WHERE <expr> }]
+
+<option> ::=
+      VIRTUAL
+    | SIZE '<size>'
+
+<name> ::= <unqualified-identifier>
+```
+
+**All of these statements will be available in experimental mode only.**
+
+The name of a cluster is an unqualified identifier. Clusters are scoped to
+a catalog, and so do not belong in the usual namespace of databases and schemas.
+For a similarly scoped object in PostgreSQL, see [publications].
+
+The `CREATE CLUSTER` statement creates a new cluster. It returns an error if
+a cluster with the specified name already exists unless the `IF NOT EXISTS`
+clause is present.
+
+The `ALTER CLUSTER` statement changes properties of an existing cluster.
+
+The `DROP CLUSTER` statement drops an existing cluster. It returns an error
+if a cluster with the specified name does not exist unless the `IF EXIST`
+clause is present. Dropping a non-empty cluster (i.e., a cluster with at least
+one index) is not permitted unless `CASCADE` is present.
+
+THe `SHOW CLUSTERS` statement lists the clusters in the system, optionally
+filtered by the provided `LIKE` pattern or `WHERE` expression, which work
+analogously to the same clauses in the `SHOW DATABASES` statement.
+
+`CREATE CLUSTER` and `ALTER CLUSTER` allow specifying the same set of cluster
+options. Initially, there are only two options:
+
+  * `VIRTUAL` requests a virtual cluster. A virtual cluster may or may not
+    correspond to actual physical resources.
+
+    We suspect that the concept of a virtual cluster is primarily useful as a
+    way to exercise the cluster logic before we can allocate true clusters, and
+    it is likely that the `VIRTUAL` option is never stabilized. But they may
+    also prove to be a useful user-facing option, in which case we can consider
+    stabilization.
+
+  * `SIZE '<size>'` specifies a cluster of a fixed size. The size is an opaque
+    string from the perspective of `materialized`, but will be validated by the
+    cloud orchestration backend.
+
+    In the initial implementation, specifying the `SIZE` option will result in
+    an error during planning (i.e., not during parsing). The cloud team will be
+    along shortly to wire up this option to turn on actual physical clusters.
+
+Specifying both `VIRTUAL` and `SIZE` options in the same `CREATE CLUSTER` or
+`ALTER CLUSTER` is an error. Specifying the same option more than once is also
+an error.
+
+### Index SQL syntax and semantics
+
+The `CREATE INDEX` statement will be extended like so to allow specifying in
+which cluster to create an index:
+
+```
+CREATE INDEX <name> ON <object> (<col>, ...) [IN CLUSTER <cluster>]
+```
+
+If the `IN CLUSTER` clause is not specified explicitly, the index is created in
+the cluster specified by the `cluster` session variable (see below). It is an
+error if that session variable does not name a valid cluster.
+
+Normalizing a `CREATE INDEX` statement will install the explicit `IN CLUSTER`
```

Comment:
Is "normalizing" known in the code as "purify"?

Reply:
Nope, it's a thing actually called normalization!

https://github.com/MaterializeInc/materialize/blob/c9fd59c8f3d3c6c606708e3d9ecb6984ded29c47/src/sql/src/normalize.rs#L121-L129